dao/OrderRepository.go

```diff
@@ -0,0 +1,118 @@
+package dao
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+)
+
+type OrderPostgres struct {
+	db *sql.DB
+}
+
+func NewOrderPostgres(db *sql.DB) *OrderPostgres {
+	return &OrderPostgres{db: db}
+}
+
+type Order struct {
+	IdDeliveryService int    `json:"delivery_service_id,omitempty"`
+	Id           int    `json:"id"`
+	IdCourier         int    `json:"courier_id,omitempty"`
+	DeliveryTime      string `json:"delivery_time,omitempty"`
+	CustomerAddress   string `json:"customer_address,omitempty"`
+	Status            string `json:"status"`
+	OrderDate         string `json:"order_date"`
+	RestaurantAddress string `json:"restaurant_address"`
+	Picked            bool   `json:"picked"`
+}
+
+type DetailedOrder struct {
+	IdDeliveryService  int    `json:"delivery_service_id,omitempty"`
+	Id            int    `json:"id"`
+	IdCourier          int    `json:"courier_id,omitempty"`
+	DeliveryTime       string `json:"delivery_time,omitempty"`
+	CustomerAddress    string `json:"customer_address,omitempty"`
+	Status             string `json:"status,omitempty"`
+	OrderDate          string `json:"order_date,omitempty"`
+	RestaurantAddress  string `json:"restaurant_address,omitempty"`
+	Picked             bool   `json:"picked,omitempty"`
+	CourierName        string `json:"name,omitempty"`
+	CourierPhoneNumber string `json:"phone_number,omitempty"`
+}
+
+func (r *OrderPostgres) AssigningOrderToCourier_InDB(order Order) error {
+	transaction, err := r.db.Begin()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Fatal would cause exit of the app. It is allowed only during startup of the application or if there is no way at all to recover. In your case I think the error isn't fatal, just error that you need to return to your user.